eclipse-collections-code-generator/src/main/resources/impl/bag/mutable/primitiveHashBag.stg

```diff
@@ -194,6 +195,19 @@ public final class <name>HashBag extends Abstract<name>Iterable implements Mutab
         this.items.forEachKeyValue(procedure);
     }
 
+    public <name>HashBag selectByOccurrences(IntPredicate predicate)
```

Comment:
I think this is fine. We usually use more concrete return types for select*() methods in concrete classes (e.g. `UnifiedSet#select()` returns `UnifiedSet`, `FastList#select()` returns `FastList` etc.). 

Think @motlin can confirm the background on this convention.